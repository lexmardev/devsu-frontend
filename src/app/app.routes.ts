import { Routes } from '@angular/router';
import { PageNotFoundComponent } from './page-not-found/page-not-found.component';
import { ProductComponent } from './modules/products/presentation/product.component';
import { productRoutes } from './modules/products/presentation/product.routes';

export const routes: Routes = [
  { path: '', component: ProductComponent, children: productRoutes },
  {
    path: '**',
    component: PageNotFoundComponent,
  },
];
