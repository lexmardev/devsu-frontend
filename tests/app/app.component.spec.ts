import { TestBed } from '@angular/core/testing';
import { AppComponent } from '../../src/app/app.component';

describe('AppComponent', () => {
  beforeEach(async () => {
    await TestBed.configureTestingModule({
      imports: [AppComponent],
    }).compileComponents();
  });

  test('should create the app', () => {
    const fixture = TestBed.createComponent(AppComponent);
    const app = fixture.componentInstance;

    expect(app).toBeTruthy();
  });

  test('must contain a router outlet', () => {
    const fixture = TestBed.createComponent(AppComponent);
    const compiled = fixture.nativeElement as HTMLElement;

    expect(compiled.querySelector('router-outlet')).toBeTruthy();
  });

  test('must match the snapshot', () => {
    const fixture = TestBed.createComponent(AppComponent);
    const compiled = fixture.nativeElement as HTMLElement;

    expect(compiled).toMatchSnapshot();
  });
});
