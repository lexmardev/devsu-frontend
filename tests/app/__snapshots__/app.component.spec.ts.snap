// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppComponent must match the snapshot 1`] = `
<div
  id="root2"
>
  <router-outlet />
  
</div>
`;
